File: db4on/core/src/core/com/db4o/foundation/HashtableIntEntry.cs
namespace com.db4o.foundation
{
	internal class HashtableIntEntry : com.db4o.foundation.DeepClone
	{
		internal int i_key;

		internal object i_object;

		internal com.db4o.foundation.HashtableIntEntry i_next;

		internal HashtableIntEntry(int a_hash, object a_object)
		{
			i_key = a_hash;
			i_object = a_object;
		}

		protected HashtableIntEntry()
		{
		}

		public virtual void AcceptKeyVisitor(com.db4o.foundation.Visitor4 visitor)
		{
			visitor.Visit(i_key);
		}

		public virtual object DeepClone(object obj)
		{
			return DeepCloneInternal(new com.db4o.foundation.HashtableIntEntry(), obj);
		}

		public virtual bool SameKeyAs(com.db4o.foundation.HashtableIntEntry other)
		{
			return i_key == other.i_key;
		}

		protected virtual com.db4o.foundation.HashtableIntEntry DeepCloneInternal(com.db4o.foundation.HashtableIntEntry
			 entry, object obj)
		{
			entry.i_key = i_key;
			entry.i_next = i_next;
			if (i_object is com.db4o.foundation.DeepClone)
			{
				entry.i_object = ((com.db4o.foundation.DeepClone)i_object).DeepClone(obj);
			}
			else
			{
				entry.i_object = i_object;
			}
			if (i_next != null)
			{
				entry.i_next = (com.db4o.foundation.HashtableIntEntry)i_next.DeepClone(obj);
			}
			return entry;
		}
	}
}

File: db4oj/core/src/com/db4o/HashtableIntEntry.java
/* Copyright (C) 2004   db4objects Inc.   http://www.db4o.com */

package com.db4o;

/**
 * 
 */
class HashtableIntEntry implements Cloneable, DeepClone {
    
    int i_key;
    Object i_object;
    HashtableIntEntry i_next;
    
    HashtableIntEntry(int a_hash, Object a_object) {
        i_key = a_hash;
        i_object = a_object;
    }

    public Object deepClone(Object obj) {
        HashtableIntEntry hie = null;
        try {
            hie = (HashtableIntEntry)clone();
        } catch (CloneNotSupportedException e) {
            // wont happen
        }
        hie.i_object = ((DeepClone)i_object).deepClone(obj);
        if(i_next != null){
            hie.i_next = (HashtableIntEntry)i_next.deepClone(obj);    
        }
        return hie;
    }
}
